util/src/main/scala/com/scalableminds/util/tools/JsonHelper.scala

```diff
@@ -119,4 +119,21 @@ object JsonHelper extends BoxImplicits with LazyLogging {
       case _ =>
         None
     }
+
+  // Sometimes play-json adds a "_type" field to the json-serialized case classes,
+  // when it thinks they can’t be distinguished otherwise. We need to remove it manually.
+  def removeGeneratedTypeFieldFromJson(jsValue: JsValue): JsValue =
+    removeKey(jsValue, "_type")
+
+  private def removeKey(jsValue: JsValue, keyToRemove: String): JsValue =
```

Comment:
the name does not suggest a recurive / deep removal of the key. I'd suggest to make this clear in the naming. e.g. `removeKeyRecursivly` or `removeKeyDeep` or something similar.